### PR TITLE
Update manuals' links

### PR DESCRIPTION
**Pull request by tssm**

### CODE REVIEW

**Comment by tssm** [comment]
I found a similar error in pill 8. If you agree I can check all the pills and update the PR

**Comment by jonringer** [comment]
I don't deal much with documentation, but fixing URLs seems like something that could be compiled into a single URL.

**Comment by jonringer** [comment]
LGTM

cc @ryantm @grahamc (documentation)

File: pills/04-basics-of-language.xml [hunk 1]
```diff
@@ -235,7 +235,7 @@
 
     <para>
       To do so, use
-      <link xlink:href="http://nixos.org/nix/manual/#idm47361539166560">recursive attribute sets</link>:
+      <link xlink:href="https://nixos.org/nix/manual/#idm140737322044432">recursive attribute sets</link>:
```
Comment (ryantm) [inline pills/04-basics-of-language.xml]:
```suggestion
      <link xlink:href="https://nixos.org/nix/manual/#idm140737322051184">recursive attribute sets</link>:
```
It seems like these `idm` anchors are not stable. This one is broken again.

File: pills/07-working-derivation.xml [hunk 2]
```diff
@@ -303,7 +303,7 @@
     </para>
     <para>
       Then we meet the
-      <link xlink:href="https://nixos.org/nix/manual/#idm140737318075200"><code>inherit</code> keyword</link>.
+      <link xlink:href="https://nixos.org/nix/manual/#idm140737322036688"><code>inherit</code> keyword</link>.
```
Comment (ryantm) [inline pills/07-working-derivation.xml]: This one is broken again too.

File: pills/08-generic-builders.xml [hunk 3]
```diff
@@ -212,7 +212,7 @@
 
     <para>
       The
-      <link xlink:href="http://nixos.org/nix/manual/#idm47361539098656">// operator</link>
+      <link xlink:href="https://nixos.org/nix/manual/#idm140737322063856">// operator</link>
```
Comment (ryantm) [inline pills/08-generic-builders.xml]: Ditto.

**Pull request merged**